### DISCUSSION TITLE
Deploy code-server on Kubernetes, providing multi-user support.

### DISCUSSION THREAD

**Discussion by torchiaf** [coder/code-server#6748] [Show and tell]
I made a simple application, https://github.com/torchiaf/code-editor.
It can be used to spin up `code-server` instances on kubernetes, one for each user.
Each user can customize their own code-server instance adding vs-code settings, extensions, git configurations and more .
Do you think it might be something worth working on / useful for people?

**Comment by code-asher** [comment]
Nice work!  I think there is some overlap with our own provisioning software (https://github.com/coder/coder) which uses Terraform and can be used with Kubernetes as well.  It supports a number of applications, either custom made or imported as modules (https://github.com/coder/modules), including code-server.

**Reply by torchiaf** [reply]
Thanks @code-asher !
Can this one https://registry.coder.com/modules/code-server create code-server instances at runtime ?
The goal of code-editor was to provide a server to a main web app, where users want to handle their own code by clicking a button and be redirected to the their fresh installed code-server view.

**Reply by code-asher** [reply]
With ours the way that would work is you could have an "Open in Coder" button that can launch a workspace based on a template that will automatically clone the repository (and it could clone settings and download extensions as well), and that workspace can have code-server as a module (and/or some other apps).

But, it is not one-click, the user has to fill out some stuff.  You can see a demo here: https://coder.com/docs/v2/latest/templates/open-in-coder  The advantage though is that you can launch code-server, or JetBrains, or use desktop VS Code, or do something else entirely.
